tools/wptrunner/wptrunner/executors/base.py

```diff
@@ -10,19 +10,23 @@
 import socket
 import sys
 from abc import ABCMeta, abstractmethod
-from typing import Any, Callable, ClassVar, Tuple, Type
+from typing import Any, Callable, ClassVar, Mapping, Tuple, Type
 from urllib.parse import urljoin, urlsplit, urlunsplit
 
 from . import pytestrunner
 from .actions import actions
 from .asyncactions import async_actions
 from .protocol import Protocol, WdspecProtocol
+from ..environment import TestEnvironment
+from ..testloader import Subsuite
+from ..wpttest import TestType, RunInfo
```

Comment:
I'd rather these be behind a `if TYPE_CHECKING` block, because when I was trying to add types to a bunch of this a while ago I just ended up suffering with circular imports as a result of this.

Reply:
Sure, done. But it feels like a circular import would indicate a layering violation, since these modules should all sit below the vendor-specific stuff in `{browsers,executors}`.